### PR TITLE
Closes #581 | Implement `vigetext` dataset

### LINKED ISSUE #581: Create dataset loader for ViGEText_17to23
Dataloader name: `vigetext/vigetext.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?vigetext

|     Dataset| vigetext |
|-------------|---|
| Description | The high-quality dataset with structured guidelines for typing LaTeX formulas in Mathematics, Physics, Chemistry, and Biology.  Objective was to cover the entire scope of the Vietnamese General Education Examination spanning from 2017 to 2023. This comprehensive approach included the challenging examinations of the years 2017 and 2018, which have been significant for nearly all Vietnamese students in recent years. It is important to highlight that the exact and unquestionably correct answers have been exclusively obtained from the Vietnamese Ministry of Education. |
| Subsets | - |
| Languages | vie |
| Tasks | Question Answering |
| License | Unknown (unknown) |
| Homepage | https://huggingface.co/datasets/uitnlp/ViGEText_17to23?row=27 |
| HF URL | https://huggingface.co/datasets/uitnlp/ViGEText_17to23?row=27 |
| Paper URL | https://dl.acm.org/doi/10.1145/3628797.3628837 |

### PR DESCRIPTION
**Pull request by chenxwh**
Closes #581.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/vigetext/vigetext.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                answer_map = {opt[0]: opt[2:].strip() for opt in question_and_options[1:]}
+                yield i, {
+                    "id": str(i),
+                    "question_id": item["id"],
+                    "document_id": "",
+                    "question": question_and_options[0],
+                    "type": "multiple_choice",
+                    "choices": [opt[2:].strip() for opt in question_and_options[1:]],  # remove A., B., ... in the options
+                    "context": "",
+                    "answer": [answer_map[item["target"]]],
```
Comment (MJonibek) [inline seacrowd/sea_datasets/vigetext/vigetext.py:149]:
```suggestion
                    "answer": [answer_map[item["target"]]],
                    "meta": {},
```

**Review by MJonibek — CHANGES_REQUESTED** [review]
@chenxwh Everything looks good, one small issue: you forgot to add `meta` key for seacrowd schema

**Review by MJonibek — APPROVED** [review]
LGTM

**Comment by holylovenia** [comment]
A friendly reminder for @akhdanfadh to review.

**Comment by akhdanfadh** [comment]
LGTM! Thank you for the dataloader @chenxwh . Merging this in a bit.

**Review by akhdanfadh — APPROVED** [review]

**Pull request merged**